packages/tooling/fast-tooling-react/src/form/custom-controls/css/color-picker-component/color-picker.ts

```diff
@@ -0,0 +1,400 @@
+import { ColorHSV, ColorRGBA64, hsvToRGB, parseColor, rgbToHSV } from "@microsoft/fast-colors";
+import { attr, DOM, observable } from "@microsoft/fast-element";
+import { isNullOrWhiteSpace } from "@microsoft/fast-web-utilities";
+import { isNumber } from "util";
+import { FormAssociatedColorPicker } from "./color-picker.form-associated";
+
+/**
+ * A Color Picker Custom HTML Element.
+ *
+ * @public
+ */
+export class ColorPicker extends FormAssociatedColorPicker 
+{
+    /**
+     * When true, the control will be immutable by user interaction. See {@link https://developer.mozilla.org/en-US/docs/Web/HTML/Attributes/readonly | readonly HTML attribute} for more information.
+     * @public
+     * @remarks
+     * HTML Attribute: readonly
+     */
+    @attr({ attribute: "readonly", mode: "boolean" })
+    public readOnly: boolean;
+    private readOnlyChanged(): void 
+    {
+        if (this.proxy instanceof HTMLElement) 
+        {
+            this.proxy.readOnly = this.readOnly;
+        }
+    }
+
+    /**
+     * Indicates that this element should get focus after the page finishes loading. See {@link https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input#htmlattrdefautofocus | autofocus HTML attribute} for more information.
+     * @public
+     * @remarks
+     * HTML Attribute: autofocus
+     */
+    @attr({ mode: "boolean" })
+    public autofocus: boolean;
+    private autofocusChanged(): void 
+    {
+        if (this.proxy instanceof HTMLElement) 
+        {
+            this.proxy.autofocus = this.autofocus;
+        }
+    }
+
+    /**
+     * Sets the placeholder value of the element, generally used to provide a hint to the user.
+     * @public
+     * @remarks
+     * HTML Attribute: placeholder
+     * Using this attribute does is not a valid substitute for a labeling element.
+     */
+    @attr
+    public placeholder: string;
+    private placeholderChanged(): void 
+    {
+        if (this.proxy instanceof HTMLElement) 
+        {
+            this.proxy.placeholder = this.placeholder;
+        }
+    }
+
+    /**
+     * Flag indicating that the color UI is visible.
+     */
+    @observable
+    public isOpen: boolean;
+
+    /**
+     * Flag indicating that the color UI is activily listening for mouse move and up events.
+     */
+    @observable
+    public isMouseActive:boolean = false;
+
+    /**
+     * Object containing all of the properties displayed in the UI.
+     */
+    @observable
+    public uiValues: ColorPickerUI = new ColorPickerUI(new ColorRGBA64(1,0,0), new ColorHSV(0,1,1));
```

Comment:
This seems intuitive, but "UI" makes me think of the *entire* UI. I assume this is specific to only the color picker UI? We don't need to update the observable, but updating the comment might make that clearer from first glance.

Reply:
I updated the comment to specify that it is for the color picker UI.